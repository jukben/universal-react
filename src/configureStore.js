import { createEpicMiddleware } from 'redux-observable';
import { applyMiddleware, combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'remote-redux-devtools';
import { createLogger } from 'redux-logger';
import { persistStore, autoRehydrate } from 'redux-persist';
import configureEpics from './configureEpics';

import viewer from './ducks/viewer';

const commonReducers = { viewer };

const configureStore = (options, storage) => {
  const { initialState, platformDeps = {}, platformEpics = [], platformReducers = {} } = options;

  const rootEpic = configureEpics({ ...platformDeps }, platformEpics);

  const reducers = combineReducers({
    ...commonReducers,
    ...platformReducers,
  });

  const middlewares = [createEpicMiddleware(rootEpic)];

  if (process.env.NODE_ENV === 'development') {
    middlewares.push(createLogger({ diff: true, collapsed: true }));
  }

  const enhancers = composeWithDevTools(applyMiddleware(...middlewares), autoRehydrate());

  const store = createStore(reducers, initialState, enhancers);

  // let's the magic happens :â€“)
  persistStore(store, { blacklist: ['ui'], storage }).purge();

  return store;
};

export default configureStore;
