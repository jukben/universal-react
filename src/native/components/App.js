// @flow

import React from 'react';
import { connect } from 'react-redux';
import { View } from 'react-native';
import { Container, Header, Spinner, Body, Title } from 'native-base';
import { appIsRunning } from '../../selectors/viewer';

import Viewer from './Viewer';

type AppProps = {
  appLoading: boolean,
};

const App = ({ appLoading }: AppProps) => (
  <Container>
    {appLoading && <Spinner style={{ flex: 1, alignSelf: 'center' }} color="blue" />}
    {!appLoading &&
      <View style={{ flex: 1 }}>
        <Header>
          <Body>
            <Title>Universal React</Title>
          </Body>
        </Header>
        <Viewer />
      </View>}
  </Container>
);

export default connect(state => ({
  appLoading: !appIsRunning(state),
}))(App);
