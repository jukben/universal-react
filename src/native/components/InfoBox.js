// @flow
import React from 'react';
import styled from 'styled-components/native';
import { Icon } from 'native-base';

type InfoBoxProps = {
  danger?: boolean, // eslint-disable-line
  text: string,
};

const InfoBoxWrapper = styled.View`
  padding-top: 5px;
  padding-bottom: 5px;
  padding-left: 10px;
  padding-right: 10px;
  margin: 10px;
  border-style: solid;
  border-color: ${({ danger }) => (danger ? '#d73a49' : '#0366d6')};
  border-width: 1;
  align-items: center;
  flex-direction: row;
`;

const InfoBoxText = styled.Text`
  color: ${({ danger }) => (danger ? '#d73a49' : '#0366d6')}
`;

const InfoBox = ({ danger = false, text }: InfoBoxProps) =>
  (<InfoBoxWrapper danger={danger}>
    <Icon
      style={{ marginRight: 5, color: danger ? '#d73a49' : '#0366d6' }}
      name={danger ? 'hand' : 'information-circle'}
    />
    <InfoBoxText danger={danger}>
      {text}
    </InfoBoxText>
  </InfoBoxWrapper>);

export default InfoBox;
