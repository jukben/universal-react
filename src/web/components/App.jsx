// @flow

import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import styled from 'styled-components';
import { appIsRunning } from '../../selectors/viewer';
import media from '../../common/helpers/media';
import Viewer from './Viewer';

const Content = styled.div`
  padding: 10px;
  max-width: 500px;
  margin: 10px auto;
  ${media.tablet`max-width: 100%;`}
`;

const AppLoader = styled.div`
  width: 100%;
  height: calc(100vh - 40px);
  font-family: Arial;
  font-size: 25px;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center; 
  align-items: center; 
`;

type AppProps = {
  appLoading: boolean,
};

const App = ({ appLoading }: AppProps) =>
  (<Router>
    <Content>
      {appLoading && <AppLoader>App is loading...</AppLoader>}
      {!appLoading &&
        <div>
          <Route path="/" component={Viewer} />
        </div>}
    </Content>
  </Router>);

export default connect(state => ({
  appLoading: !appIsRunning(state),
}))(App);
