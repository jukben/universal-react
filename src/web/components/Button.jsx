import styled from 'styled-components';

const Button = styled.button`
    background: #eff3f6;
    background-image: linear-gradient(-180deg, #fafbfc 0%, #eff3f6 90%);
    background-position: -1px -1px;
    background-size: 110% 110%;
    border: 1px solid rgba(27,31,35,0.2);
    border-radius: 0.25em;
    padding: 10px;
    text-transform: uppercase;
    outline: none;
    cursor: pointer;
    display: flex;
    align-items: center;

    &:disabled {
        opacity: .7;
        cursor: not-allowed;
    }

    &:hover:not(:disabled) {
        background: #e6ebf1;
        background-image: linear-gradient(-180deg, #f0f3f6 0%, #e6ebf1 90%);
        border-color: rgba(27,31,35,0.35);
    } 
`;

export default Button;
