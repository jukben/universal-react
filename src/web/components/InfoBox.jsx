// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {
  danger?: boolean, //eslint-disable-line
  children?: string, //eslint-disable-line
};

const InfoBoxStyled = styled.div`
  padding: 10px;
  font-family: Arial;
  border: solid 1px ${({ danger }) => (danger ? '#d73a49' : '#0366d6')}
  color: ${({ danger }) => (danger ? '#d73a49' : '#0366d6')};
  border-radius: 3px;
`;

const InfoBox = ({ danger = false, children }: Props) => (
  <InfoBoxStyled danger={danger}>
    {danger
      ? <span role="img" aria-label="stop">✋</span>
      : <span role="img" aria-label="points up">☝</span>}
    {' '}
    {children}
  </InfoBoxStyled>
);

export default InfoBox;
