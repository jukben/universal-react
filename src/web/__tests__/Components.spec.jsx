import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../configureStore';

import storeFetchedMock from '../../__test__/state.fetched.json';

import App from '../components/App';
import Button from '../components/Button';
import InfoBox from '../components/InfoBox';

const storeEmpty = configureStore({
  initialState: {},
  platformDeps: {},
  platformEpics: [],
  platformReducers: {},
});

const storeFetched = configureStore({
  initialState: storeFetchedMock,
  platformDeps: {},
  platformEpics: [],
  platformReducers: {},
});

const injected = (Component, store = storeEmpty) => <Provider store={store}>{Component}</Provider>;

describe('Web components with empty store', () => {
  it('App renders correctly', () => {
    const tree = renderer.create(injected(<App />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Button renders correctly', () => {
    const tree = renderer.create(injected(<Button />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('InfoBox with text renders correctly', () => {
    const tree = renderer.create(injected(<InfoBox>Text</InfoBox>)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('InfoBox (danger) with text renders correctly', () => {
    const tree = renderer.create(injected(<InfoBox danger>Text</InfoBox>)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Web components with mocked store (fetched data)', () => {
  it('App renders correctly', () => {
    const tree = renderer.create(injected(<App />, storeFetched)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Button renders correctly', () => {
    const tree = renderer.create(injected(<Button />, storeFetched)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('InfoBox with text renders correctly', () => {
    const tree = renderer.create(injected(<InfoBox>Text</InfoBox>)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('InfoBox (danger) with text renders correctly', () => {
    const tree = renderer.create(injected(<InfoBox danger>Text</InfoBox>)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
