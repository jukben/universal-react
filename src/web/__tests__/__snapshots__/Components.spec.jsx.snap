// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Web components with empty store App renders correctly 1`] = `
<div
  className="cAkaVW"
>
  <div
    className="hmDDgf"
  >
    App is loading...
  </div>
</div>
`;

exports[`Web components with empty store Button renders correctly 1`] = `
<button
  className="cKltRP"
/>
`;

exports[`Web components with empty store InfoBox (danger) with text renders correctly 1`] = `
<div
  className="dFzGre"
>
  <span
    aria-label="stop"
    role="img"
  >
    ✋
  </span>
   
  Text
</div>
`;

exports[`Web components with empty store InfoBox with text renders correctly 1`] = `
<div
  className="fhvbIR"
>
  <span
    aria-label="points up"
    role="img"
  >
    ☝
  </span>
   
  Text
</div>
`;

exports[`Web components with mocked store (not logged) App renders correctly 1`] = `
<div
  className="cAkaVW"
>
  <div />
</div>
`;

exports[`Web components with mocked store (not logged) Button renders correctly 1`] = `
<button
  className="cKltRP"
/>
`;

exports[`Web components with mocked store (not logged) InfoBox (danger) with text renders correctly 1`] = `
<div
  className="dFzGre"
>
  <span
    aria-label="stop"
    role="img"
  >
    ✋
  </span>
   
  Text
</div>
`;

exports[`Web components with mocked store (not logged) InfoBox with text renders correctly 1`] = `
<div
  className="fhvbIR"
>
  <span
    aria-label="points up"
    role="img"
  >
    ☝
  </span>
   
  Text
</div>
`;
